### ISSUE TITLE
Not all specified Ubuntu operatingsystemrelease captured 

### ISSUE THREAD

**Issue by snesbittsea** [mcanevet/rspec-puppet-facts#22]
I've run into unexpected behavior with respect to Ubuntu operatingsystemreleases. Specifically, rspec-puppet-facts doesn't appear to process all specified minor releases.

Here's the relevant section of metadata.json:
```
      "operatingsystem_support": [
    {
      "operatingsystem": "Ubuntu",
      "operatingsystemrelease": [
        "12.04",
        "14.04",
        "14.10",
        "15.04",
        "15.10"
      ]
    }
  ]
```

I would expect rspec-puppet-facts to iterate through each operatingsystemrelease allowing me to test for run tests for 12.04, 14.04, 14.10, 15.04, 15.10. It's not. Instead I get iterations only on 12.04, 14.10, 15.10. No opportunity to test 14.04 or 15.04 is provided.

It kinda acts like rspec-puppet-facts is doing some kind of filtering on major version (12, 14, 15) such that only one minor version (the largest) is kept. Is this expected behavior?

**Comment by bodgit** [comment]
I've noticed similar behaviour with OpenBSD releases. If I have 5.7, 5.8 and 5.9 I just get one release (not entirely such which one wins) which is presumably matched against 5.

**Comment by mcanevet** [comment]
@snesbittsea I added a unit test to prove the bug and I'm trying to fix it on all versions of facter (not easy on version < 2.2)
@bodgit the problem for OpenBSD is quite different as facter returns 5 as operatingsystemmajrelease for OpenBSD while it probably should be `5.7`, `5.8` and `5.9`.

**Comment by mcanevet** [comment]
Should be fixed in 1.6.1
